examples/MQ/pixelDetector/src/PixelDigiSource.cxx

```diff
@@ -134,10 +126,10 @@ Int_t PixelDigiSource::ReadEvent(UInt_t i)
 
 Bool_t PixelDigiSource::ActivateObject(TObject** obj, const char* BrName)
 {
-    if (strcmp(BrName, "PixelDigis"))
-        *obj = (TObject*)fDigis;
-    else if (strcmp(BrName, "EventHeader."))
-        *obj = (TObject*)fEventHeader;
+    if (strcmp(BrName, "PixelDigis") == 0)
+        *obj = &fDigis;
+    else if (strcmp(BrName, "EventHeader.") == 0)
+        *obj = &fEventHeader;
```

Comment:
The previous `strcmp` usage here looks like a bug to me! I wonder, why this worked at all before? Or is it not called at all?

Reply:
Probably a bug, we are not testing this anywhere.

Reply:
Just a wild guess:

The `Register` that happens in ::Init seeds FairRootManager enough, so that GetObject will not try to Activate?
https://github.com/FairRootGroup/FairRoot/blob/cfdc0e708247bf545f904f7b0c510e693bcaf978/examples/MQ/pixelDetector/src/PixelDigiSource.cxx#L59